### ISSUE TITLE
getParameter() super slow on Macs with nvidia GPUs

### ISSUE THREAD

**Issue by shaojingli** [uber/deck.gl#472]
For pickable layers, getParameter() calls in getBlendMode() are super slow. Seems to be a GPU/platform specific issue that we can solve directly so we need to have a workaround.

Does the layer in our chicago building demo actually need to be pickable?

<img width="1219" alt="screen shot 2017-03-27 at 4 34 02 pm" src="https://cloud.githubusercontent.com/assets/21322704/24382473/4fb678fc-130b-11e7-8a68-c9f94ab1729e.png">

**Comment by ibgreen** [comment]
This will be solved for all similar cases by the new state manager in luma.gl v4. I would be OK with a simpler fix here for now.

**Comment by shaojingli** [comment]
> This will be solved for all similar cases by the new state manager in luma.gl v4. I would be OK with a simpler fix here for now.

Yes. That will solve the problem once and for all. Currently, we are clearing picking framebuffers, initialize blending mode and set up blend equations even when we don't have any pickable layers. For this specific issue, we might want to patch the code to bypass those.

**Comment by shaojingli** [comment]
There seems to be some driver issue around the NVIDIA GPUs used by Mac. 

One symptom is that the 64-bit building layer rendered in our Chicago building demo is super slow. This can be mitigate by turning off 64-bit. However, I was regularly hit by GPU kernel panic when looking at this specific demo. 

This is not the only GeoJsonLayer we have in our demos. I believe the ArcLayer demo and TripsLayer demo also have GeoJsonLayer inside. Not sure what makes this happen.

**Comment by ibgreen** [comment]
> There seems to be some driver issue around the NVIDIA GPUs used by Mac.

We did testing on this for v3 on nVidia when we released right? Can we redo that? E.g. if you run the ChoroplethExample of the current github pages, how is performance on the nVidia Mac? Does it also exhibit low FPS / Kernel Panic.

Always good to know if are we dealing with a regression. If it the issue is not present on the older layer, the v4 fp64 library and the v4 SolidPolygonLayer (I assume this is what is rendering the polys in the GeoJson example) are obvious suspects. Also could make sense to double check uniforms and attribute values in the luma log, as I recall the nVidia GPUs can't handle `NaN`s.
